protocol-units/da/movement/protocol/light-node/src/sequencer.rs

```diff
@@ -163,13 +164,29 @@ where
 	}
 
 	/// Submits blocks to the pass through.
+	///
+	/// Note: if you change the block submission architecture to an eventually consistent submission pattern, you will want to revert this to a sequential submission.
+	/// Currently, all of the blobs can just be sent through because we are relying on Celestia for order.
 	async fn submit_blocks(&self, blocks: Vec<Block>) -> Result<(), anyhow::Error> {
-		for block in blocks {
+		let futures = blocks.into_iter().map(|block| async {
 			let data: InnerSignedBlobV1Data<C> = block.try_into()?;
 			let blob = data.try_to_sign(&self.pass_through.signer).await?;
 			self.pass_through.da.submit_blob(blob.into()).await?;
-		}
+			Ok::<(), anyhow::Error>(())
+		});
+
+		join_all(futures).await.into_iter().collect::<Result<(), _>>()?;
+		Ok(())
+	}
 
+	/// Collapses the back-pressured blocks into a single block and submits it.
+	///
+	/// todo: mark not pub once this is actually used
+	pub async fn submit_collapsed_blocks(&self, blocks: Vec<Block>) -> Result<(), anyhow::Error> {
```

Comment:
Will we need this in the future?

Reply:
I understand the fix, this is not part of it though. What I'm asking is, if we'll likely need this currently unused method in the future, or is it another piece of dead code that will incur cognitive costs on anyone who needs to work with the code base.